tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -0,0 +1,118 @@
+import sys
+import os
+import subprocess
+import json
+import logging
+
+from parameterized import parameterized
+
+from samcli.yamlhelper import yaml_parse
+from .build_integ_base import BuildIntegBase
+
+
+LOG = logging.getLogger(__name__)
+
+
+class TestBuildCommand_PythonFunctions(BuildIntegBase):
+
+    EXPECTED_FILES_GLOBAL_MANIFEST = set()
+    EXPECTED_FILES_PROJECT_MANIFEST = {'__init__.py', 'main.py', 'numpy', 'cryptography', "jinja2",
+                                       'requirements.txt'}
+
+    FUNCTION_LOGICAL_ID = "PythonFunction"
+
+    @parameterized.expand([
+        ("python2.7", False),
+        ("python3.6", False)
+
+        # FIXME: Turning off Container build tests until official Docker Container is published
+        # ("python2.7", "use_container"),
+        # ("python3.6", "use_container"),
+    ])
+    def test_with_default_requirements(self, runtime, use_container):
+
+        # Don't run test on wrong Python versions
+        py_version = self._get_python_version()
+        if py_version != runtime:
+            self.skipTest("Current Python version '{}' does not match Lambda runtime version '{}'".format(py_version,
+                                                                                                          runtime))
+
+        overrides = {"PythonRuntime": runtime}
+        cmdlist = self.get_command_list(use_container=use_container,
+                                        parameter_overrides=overrides)
+
+        LOG.info("Running Command: {}", cmdlist)
+        process = subprocess.Popen(cmdlist, cwd=self.working_dir)
+        process.wait()
+
+        self._verify_built_artifact(self.default_build_dir, self.FUNCTION_LOGICAL_ID,
+                                    self.EXPECTED_FILES_PROJECT_MANIFEST)
+
+        expected = {
+            "pi": "3.14",
+            "jinja": "Hello World"
+        }
+        self._verify_invoke_built_function(self.built_template,
+                                           self.FUNCTION_LOGICAL_ID,
+                                           self._make_parameter_override_arg(overrides),
+                                           expected)
+
+    def _verify_invoke_built_function(self, template_path, function_logical_id, overrides, expected_result):
+        LOG.info("Invoking built function '{}'", function_logical_id)
+
+        cmdlist = [self.cmd, "local", "invoke", function_logical_id, "-t", str(template_path), "--no-event",
+                   "--parameter-overrides", overrides]
+
+        process = subprocess.Popen(cmdlist, stdout=subprocess.PIPE)
+        process.wait()
+
+        process_stdout = b"".join(process.stdout.readlines()).strip().decode('utf-8')
+        print(process_stdout)
```

Comment:
we can probably remove the prints?

Reply:
I do want to have prints for debugging. pytest prints them only if your test fails